src/nvim/main.c

```diff
@@ -921,6 +921,10 @@ static void remote_request(mparm_T *params, int remote_args, char *server_addr,
       os_errmsg(connect_error);
       os_errmsg("\n");
       os_exit(1);
+    } else if (strequal(server_addr, os_getenv("NVIM"))) {
```

Comment:
`strequal` is a fragile way to check if the server is the same. But it probably catches most accidents.

Ideally we could check some sort of server identifier. https://github.com/neovim/neovim/pull/1758 would be an obvious way, but until then another way is to check the `stdpath('run')` directory. 

But that isn't a blocker for this PR.

Reply:
i'm not very familiar with the codebase, why is strequal fragile?

Reply:
Because different forms of a server address may resolve to the same server. For example `/foo//bar` is the same server as `/foo/bar` , yet `strequal()` thinks they are different.

Reply:
I see, that makes sense